objects.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+/*
+1. Picture gallery
+*/
+function getLeftValueForImage(images, widths, imageIndex) {
+    let result = 0;
+    for (let i =0 ; i < imageIndex; i++) {
+        result-=widths[i]
+    }
+    return result;
+}
+
+const images = ['1.jpg', '2.jpg', '3.jpg'];
+const widths = [200, 300, 400];
+
+let leftValue = getLeftValueForImage(images, widths, 0);
+console.log(`Left value should be 0: ${leftValue}`);
+leftValue = getLeftValueForImage(images, widths, 1);
+console.log(`Left value should be -200: ${leftValue}`);
+leftValue = getLeftValueForImage(images, widths, 2);
+console.log(`Left value should be -500: ${leftValue}`);
+leftValue = getLeftValueForImage(images, widths, 3);
+console.log(`Left value should be -900: ${leftValue}`);
+
+function compareCars(car1, car2) {
+    if (car1.speed > car2.speed) {
+        return car1;
+    }
+    return car2;
+}
+
+/*
+2. Fastest car
+*/
+function getFastestCar(cars) {
+    return cars.reduce(compareCars);
+}
+
+const cars = [ { name: 'c1', color: 'red', speed: 40 }, { name: 'c2', color: 'blue', speed: 20 }];
+
+const fastestCar = getFastestCar(cars);
+console.log(fastestCar.name) // should print c1
+
+/*
+3. tic tac toe winner
+*/
+
+function checkLine(board, startX, startY, xInc,yInc) {
```

Comment:
The solution seems to be simple, yet here is an issue that a lot of developers have to deal with:
I have no idea what `xInc & yInc` are 😆
Can you explain what they are and give them more descriptive names?